io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -334,31 +337,69 @@ schedule !thread@Thread{
 
     NewTimeout d k ->
       {-# SCC "schedule.NewTimeout.2" #-} do
-      !tvar  <- execNewTVar nextVid
-                            (Just $ "<<timeout-state " ++ show (unTimeoutId nextTmid) ++ ">>")
-                            TimeoutPending
-      !tvar' <- execNewTVar (succ nextVid)
-                            (Just $ "<<timeout " ++ show (unTimeoutId nextTmid) ++ ">>")
-                            False
-      let !expiry  = d `addTime` time
-          !t       = Timeout tvar tvar' nextTmid
-          !timers' = PSQ.insert nextTmid expiry (TimerVars tvar tvar') timers
-          !thread' = thread { threadControl = ThreadControl (k t) ctl }
-      !trace <- schedule thread' simstate { timers   = timers'
-                                          , nextVid  = succ (succ nextVid)
-                                          , nextTmid = succ nextTmid }
+      tvar  <- execNewTVar nextVid
+                           (Just $ "<<timeout-state " ++ show (unTimeoutId nextTmid) ++ ">>")
+                           TimeoutPending
+      let expiry  = d `addTime` time
+          t       = Timeout tvar nextTmid
+          timers' = PSQ.insert nextTmid expiry (Timer tvar) timers
+          thread' = thread { threadControl = ThreadControl (k t) ctl }
+      trace <- schedule thread' simstate { timers   = timers'
+                                         , nextVid  = succ nextVid
+                                         , nextTmid = succ nextTmid }
```

Comment:
Were bangs removed by a rebase or do we actually want to remove them?

Reply:
Done